apm-server: Remove server information endpoint `POST /` support (#647)
elastic/docs-content: release-notes/breaking-changes/elastic-apm.md

% **Action**<br> Steps for mitigating deprecation impact.
% ::::

## 9.0.0 [elastic-apm-900-breaking-changes]
**Release date:** March 25, 2025

% ::::{dropdown} Title of breaking change 
% Description of the breaking change.
% For more information, check [PR #](PR link).
% **Impact**<br> Impact of the breaking change.
% **Action**<br> Steps for mitigating deprecation impact.
% ::::

> ::::{dropdown} Change server information endpoint / to only accept GET and HEAD requests

Author: In release-notes markdown, there's a comment
> % Release notes include only features, enhancements, and fixes. Add breaking changes, deprecations, and known issues to the applicable release notes sections. 
% For each new version section, include the Elastic APM and Kibana changes. 

[to reviewer] so we're not supposed to mention breaking changes in release notes changelog, but only in this breaking changes section from now on?

Reviewer: From @KOTungseth:

>Correct—the new structure separates breaking changes from the release notes changelog. Moving forward, breaking changes should only be documented in the "Breaking Changes" section, not in the general release notes. Since [breaking changes appear in the left nav in the release notes section of our new IA](https://docs-v3-preview.elastic.dev/elastic/docs-content/tree/main/release-notes/) [this is just a subset of release-note-related content as we wait for the dev work to be done related to assembling docs from all repos together], users can easily navigate and view these pages. 

@KOTungseth can you share a link to more details about the release note strategy with @carsonip when it's ready? 🙏

Author: @colleenmcginnis sgtm. I believe this PR aligns with the strategy. Feel free to review and merge it.

> ::::{dropdown} Change server information endpoint / to only accept GET and HEAD requests

Reviewer: It's a bit confusing in the dropdown title since we can't format as `/`.

<img width="802" alt="image" src="https://github.com/user-attachments/assets/7d8fd08c-a9a3-4e82-8c26-c72fd7c0300d" />

What do you think about this instead? 

```suggestion
::::{dropdown} Change server information endpoint "/" to only accept GET and HEAD requests
```

> This will surface any agent misconfiguration causing data to be sent to `/` instead of the correct endpoint, e.g. `/v1/traces` for OTLP/HTTP.

Reviewer: ```suggestion
This will surface any agent misconfiguration causing data to be sent to `/` instead of the correct endpoint (for example, `/v1/traces` for OTLP/HTTP).
```

For more information, check [PR #15976](https://github.com/elastic/apm-server/pull/15976).

> **Impact**<br> Any methods other than GET and HEAD to `/` will return HTTP 405 Method Not Allowed.
> **Action**<br> Update any existing usage of e.g. `POST /` to `GET /`.

Reviewer: ```suggestion
**Impact**<br> Any methods other than `GET` and `HEAD` to `/` will return HTTP 405 Method Not Allowed.
**Action**<br> Update any existing usage, for example, update `POST /` to `GET /`.
```

% For more information, check [PR #](PR link).
% **Impact**<br> Impact of the breaking change.
% **Action**<br> Steps for mitigating deprecation impact.
% ::::

> % ::::{dropdown} Change server information endpoint `/` to only accept GET and HEAD requests

Author: TODO: this may move back to apm-server

Author: No need to wait. We can make the change before the move.
